### PR TITLE
Add credential and new api

### PR DESCRIPTION
**Pull request by antho1404**
dependency: https://github.com/mesg-foundation/mesg-js/pull/143

This is adding the credential system to fix issues with the deployment of the services

It also uses the new API `service.hash` and `service.exists` to fix the issue with the wrong error catching.

### CODE REVIEW

**Comment by antho1404** [comment]
The build is failing because of the deletion of the service.delete api, this is fixed in https://github.com/mesg-foundation/cli/pull/209

**Comment by NicolasMahe** [comment]
@antho1404 can you fix conflict.

**Comment by antho1404** [comment]
fixed

File: src/commands/service/create.ts [hunk 1]
```diff
@@ -27,14 +27,24 @@ export default class ServiceCreate extends Command {
...
-    const resp = await this.api.service.create(JSON.parse(args.DEFINITION), credential)
-    if (!resp.hash) { throw new Error('invalid response') }
-    this.spinner.stop(base58.encode(resp.hash))
+    const service = JSON.parse(args.DEFINITION)
+
+    const {hash} = await this.api.service.hash(service)
+    if (!hash) throw new Error('invalid hash')
+    const {exists} = await this.api.service.exists({hash})
+    if (exists) {
+      this.warn('Service already exists')
```
Comment (NicolasMahe) [inline src/commands/service/create.ts]: why not checking if service exist and not simply output the error from service.create?

Comment (antho1404) [reply src/commands/service/create.ts]: done

**Pull request merged**